packages/synthetics-sdk-broken-links/test/unit/broken_links_main.spec.ts

```diff
@@ -0,0 +1,163 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import { expect, use } from 'chai';
+import chaiExclude from 'chai-exclude';
+use(chaiExclude);
+
+import {
+  BrokenLinksResultV1_BrokenLinkCheckerOptions,
+  BrokenLinksResultV1_BrokenLinkCheckerOptions_LinkOrder,
+  BrokenLinksResultV1_SyntheticLinkResult,
+  ResponseStatusCode,
+  ResponseStatusCode_StatusClass,
+} from '@google-cloud/synthetics-sdk-api';
+import {
+  runBrokenLinks,
+  BrokenLinkCheckerOptions,
+} from '../../src/broken_links_main';
+const path = require('path');
+
+describe('runBrokenLinks', async () => {
```

Comment:
I would test that branching logic that is specifically found only within runBrokenLinks is captured in these tests. Understandably, we cant test all the logic in the functions that runBrokenLinks calls, but there's behavior occurring between each step that doesnt seem to be completely captured across the board.

Reply:
I ran the code coverage. the only unchecked branch was the wait_for_selector but I added a test for that.